crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -27,19 +31,47 @@ use bevy_utils::EntityHashMap;
 /// }
 ///
 /// impl MapEntities for Spring {
-///     fn map_entities(&mut self, entity_mapper: &mut EntityMapper) {
-///         self.a = entity_mapper.get_or_reserve(self.a);
-///         self.b = entity_mapper.get_or_reserve(self.b);
+///     fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M) {
+///         self.a = entity_mapper.map(self.a);
+///         self.b = entity_mapper.map(self.b);
 ///     }
 /// }
 /// ```
 ///
 pub trait MapEntities {
     /// Updates all [`Entity`] references stored inside using `entity_mapper`.
-    ///
-    /// Implementors should look up any and all [`Entity`] values stored within and
-    /// update them to the mapped values via `entity_mapper`.
-    fn map_entities(&mut self, entity_mapper: &mut EntityMapper);
+    fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M);
+}
+
+/// Any implementor of this trait knows how to map an [`Entity`] into another [`Entity`].
+///
+/// Usually this is done by using a [`EntityHashMap<Entity, Entity>`] to map the [`Entity`] references.
+/// This can be used to map [`Entity`] references from one [`World`] to another.
+pub trait Mapper {
```

Comment:
Not quite sure if `Mapper` is a good name...
Personally I would suggest to rename it into `EntityMapper` and current `EntityMapper` struct into `WorldEntityMapper`.
@alice-i-cecile ?

Reply:
I like that!

Reply:
Yep that sounds lovely.